### PR TITLE
MM 42449 Enterprise Trial Requests

### PR DESCRIPTION
**Pull request by munish7771**
#### Summary
TL;DR
1. Added models for enterprise trial requests to sync to salesforce.
2. Refactored existing code wrt professional subscriptions

### CODE REVIEW

File: transform/snowflake-dbt/models/hightouch/freemium/enterprise_trial_requests/etr_campaignmember_insert.sql [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        COALESCE(
+            existing_leads.dwh_external_id__c,
+            UUID_STRING(
+                '78157189-82de-4f4d-9db3-88c601fbc22e',
+                customers_with_freemium_subs.customer_id || customers_with_freemium_subs.email)
+        ) AS lead_external_id,
+        '7016u0000002OHjAAM' as sandbox_campaign_id,
+        '7013p000001U2GbAAK' as campaign_id,
+        'Responded' as campaignmember_status
+    FROM {{ ref('customers_with_freemium_subs_without_filter') }} as customers_with_freemium_subs
```
Comment (catalintomai) [inline transform/snowflake-dbt/models/hightouch/freemium/enterprise_trial_requests/etr_campaignmember_insert.sql:45]: why not name `as customers_with_freemium_subs` -> `as customers_with_freemium_subs_without_filter` to be consistent?

File: transform/snowflake-dbt/models/blapi/customers_with_freemium_subs_without_filter.sql [hunk 2]
```diff
@@ -0,0 +1,159 @@
+{{config({
```
Comment (catalintomai) [inline transform/snowflake-dbt/models/blapi/customers_with_freemium_subs_without_filter.sql:1]: add a comment on how this file differs from `customers_with_freemium_subs.sql` ?

File: transform/snowflake-dbt/models/hightouch/freemium/enterprise_trial_requests/etr_campaignmember_insert.sql [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    from {{ ref('campaignmember') }}
+    where campaignmember.campaignid = '7016u0000002OHjAAM'
+    -- where campaignmember.campaignid = '7013p000001U2GbAAK'
+), existing_leads as (
+    select
+        lead.sfid,
+        lead.email,
+        lead.dwh_external_id__c,
+        row_number() over (partition by lead.email order by createddate desc) as row_num
+    from {{ ref('lead') }}
```
Comment (catalintomai) [inline transform/snowflake-dbt/models/hightouch/freemium/enterprise_trial_requests/etr_campaignmember_insert.sql:22]: in another file, you've renamed `lead` to `freemium_leads` - should we be consistent naming-wise across these files?

File: transform/snowflake-dbt/models/hightouch/freemium/professional_subscription/freemium_contact.sql [hunk 4]
```diff
@@ -9,7 +9,7 @@ WITH existing_lead AS (
     SELECT
         lead.id,
         lead.email
-    FROM {{ ref('lead') }}
+    FROM {{ ref('freemium_leads') }}
```
Comment (catalintomai) [inline transform/snowflake-dbt/models/hightouch/freemium/professional_subscription/freemium_contact.sql:12]: can you detail/document why this change?

Comment (munish7771) [reply transform/snowflake-dbt/models/hightouch/freemium/professional_subscription/freemium_contact.sql:12]: it was dev code that i accidentally pushed, removed it now.

File: transform/snowflake-dbt/models/blapi/customers_with_freemium_subs_without_filter.sql (hunk 2 above)
Comment (munish7771) [reply transform/snowflake-dbt/models/blapi/customers_with_freemium_subs_without_filter.sql]: I have removed this model entirely. Added comment for the new one.

File: transform/snowflake-dbt/models/hightouch/freemium/enterprise_trial_requests/etr_campaignmember_insert.sql (hunk 1 above)
Comment (munish7771) [reply transform/snowflake-dbt/models/hightouch/freemium/enterprise_trial_requests/etr_campaignmember_insert.sql]: this was part of previous logic, which I have updated in new models.

**Review by sri-byte — APPROVED** [review]
